rust-runtime/aws-smithy-runtime/src/client/orchestrator/http.rs

```diff
@@ -33,3 +41,41 @@ pub(crate) async fn read_body(response: &mut HttpResponse) -> Result<(), <SdkBod
 
     Ok(())
 }
+
+#[derive(Debug)]
+pub(crate) struct SensitiveOutput;
+impl Storable for SensitiveOutput {
+    type Storer = StoreReplace<Self>;
+}
+
+/// An interceptor to inject a marker `SensitiveOutput`, causing wire logging of a response body
+/// to be disabled.
+#[derive(Debug)]
+pub struct SensitiveOutputInterceptor;
+impl Interceptor for SensitiveOutputInterceptor {
+    fn name(&self) -> &'static str {
+        "SensitiveOutputInterceptor"
+    }
+
+    fn modify_before_deserialization(
+        &self,
+        _context: &mut BeforeDeserializationInterceptorContextMut<'_>,
+        _runtime_components: &RuntimeComponents,
+        cfg: &mut ConfigBag,
+    ) -> Result<(), BoxError> {
+        cfg.interceptor_state().store_put(SensitiveOutput);
```

Comment:
Should the operation runtime plugin add this to config rather than it being a separate interceptor?

Reply:
If the operation runtime plugin added it, then the question would be how can existing call sites of a generic client in `aws_config` ([example](https://github.com/awslabs/smithy-rs/blob/2c27834f90b0585dba60606f9da6246b341227d6/aws/rust-runtime/aws-config/src/http_credential_provider.rs#L48)) express they want to redact response bodies, once the middleware has been removed?

Previously with the middleware, they were able to express that through the implementation of the [sensitive](https://github.com/awslabs/smithy-rs/blob/75461b9db9a68131efcfaee05ae8440548fed945/aws/rust-runtime/aws-config/src/imds/client.rs#L284-L286) method of the `ParseStringResponse` trait. But that trait is gone now in the orchestrator.

With the orchestrator, I believe those call sites can only specify redacting response bodies through an interceptor.

Reply:
A runtime plugin can also add config, so I don't think it's limited to just interceptors. The `SensitiveOutput` config value could still be available for aws-config to add.

Reply:
Addressed in e6c60da